Source/WarpX.cpp

```diff
@@ -429,10 +429,49 @@ WarpX::ReadParameters ()
             fine_tag_hi = RealVect{hi};
         }
 
+        // select which particle comps to write
+        {
+            std::map<std::string, int> particle_var_map;
+            particle_var_map["particle_weight"]         = 0;
+            particle_var_map["particle_momentum_x"]     = 1;
+            particle_var_map["particle_momentum_y"]     = 2;
+            particle_var_map["particle_momentum_z"]     = 3;
+            particle_var_map["particle_efield_x"]       = 4;
+            particle_var_map["particle_efield_y"]       = 5;
+            particle_var_map["particle_efield_z"]       = 6;
+            particle_var_map["particle_bfield_x"]       = 7;
+            particle_var_map["particle_bfield_y"]       = 8;
+            particle_var_map["particle_bfield_z"]       = 9;
+#ifdef WARPX_STORE_OLD_PARTICLE_ATTRIBS
+            particle_var_map["particle_x_old"]          = 10;
+            particle_var_map["particle_y_old"]          = 11;
+            particle_var_map["particle_z_old"]          = 12;
+            particle_var_map["particle_momentum_x_old"] = 13;
+            particle_var_map["particle_momentum_y_old"] = 14;
+            particle_var_map["particle_momentum_z_old"] = 15;
```

Comment:
It seems that this uses the same ordering as here:
https://github.com/ECP-WarpX/WarpX/blob/master/Source/Particles/WarpXParticleContainer.H#L9
Could we use the same names, and replace the hard-coded numbers with the corresponding enum:
```
particle_var_map["w"]         = PIdx::w;
particle_var_map["ux"]         = PIdx::ux;
etc...
```
Also, should the `particle_var_map` be defined in `WarpXParticleContainer.H`? 
This would help ensuring that we update it whenever we update the `PIdx`, as in the case here:
https://github.com/ECP-WarpX/WarpX/pull/69/files#diff-09902398ce079f01037240e84af93eea

Reply:
Just to know, is there a reason for not using an `enum` or `enum class` here?
Is it because we may have to extend it when `#ifdef WARPX_STORE_OLD_PARTICLE_ATTRIBS`?

Reply:
I think that `PIdx::w`, `PIdx::ux` is an `enum`, if I am correct.

Reply:
Yes, sorry, I meant `particle_var_map`could be an enum too

Reply:
Ah, ok, yes.
I think that the issue is that the user provides strings in the input file (through `warpx.particle_plot_vars`), so there needs to be a conversion from string to integers (i.e. a dictionary, or "map" in C++ lingo).
But maybe I am wrong ; @atmyers could confirm whether this is why he had to use a `map`.

Reply:
Yes, that is why we need a map. I went ahead and moved this to `WarpXParticleContainer.H` and changed the names.